### PR TITLE
Seed password protected

### LINKED ISSUE #49: Allow derive key using a password
See https://github.com/Zondax/filecoin-rs/pull/43#discussion_r382555976

`derive_child_key` could have a an extra argument to take a password and decrypt the mnemonic seed.

### PR DESCRIPTION
**Pull request by rllola**
closes #49;

- [x] Update native lib;
- [x] Update wasm lib;
- [x] Update json-rpc (password has empty string as default value);
- [x] Update documentation;
- [x] Update FFI examples ?

### CODE REVIEW

**Comment by c410-f3r** [comment]
LGTM

File: examples/wasm_node/test/test.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ describe('Key generation / derivation', function() {
 
     it('Key Derive Invalid Path', () => {
         assert.throws(
-            () => signer_wasm.key_derive(EXAMPLE_MNEMONIC, "m/44'/461'/a/0/1"),
+            () => signer_wasm.key_derive(EXAMPLE_MNEMONIC, "m/44'/461'/a/0/1", ""),
```
Comment (jleni) [inline examples/wasm_node/test/test.js:102]: Can we add a test with a "password" to be sure it actually works and changes the result?

File: signer/src/lib.rs [hunk 2]
```diff
@@ -108,17 +108,18 @@ pub fn key_generate_mnemonic() -> Result<Mnemonic, SignerError> {
...
 /// * `mnemonic` - A string containing a 24-words English mnemonic
 /// * `path` - A string containing a derivation path
-pub fn key_derive(mnemonic: &str, path: &str) -> Result<ExtendedKey, SignerError> {
+/// * `password` - Password to decrypt seed, if none use and empty string (e.g "")
+pub fn key_derive(mnemonic: &str, path: &str, password: &str) -> Result<ExtendedKey, SignerError> {
     let mnemonic = bip39::Mnemonic::from_phrase(&mnemonic, Language::English)
         .map_err(|err| SignerError::GenericString(err.to_string()))?;
 
-    let seed = Seed::new(&mnemonic, "");
+    let seed = Seed::new(&mnemonic, password);
```
Comment (jleni) [inline signer/src/lib.rs:122]: Can we create a test for some known password and compare with some test vector source from some other recognized reference?

File: signer/src/lib.rs [hunk 3]
```diff
@@ -383,7 +384,7 @@ mod tests {
     fn derive_key() {
         let mnemonic = "equip will roof matter pink blind book anxiety banner elbow sun young";
 
-        let extended_key = key_derive(mnemonic, "m/44'/461'/0/0/0").unwrap();
+        let extended_key = key_derive(mnemonic, "m/44'/461'/0/0/0", "").unwrap();
```
Comment (jleni) [inline signer/src/lib.rs:387]: Similarly, here.. let's add some test using known values

**Comment by jleni** [comment]
Try rebasing or merging dev into this PR. It should fix most of the CI issues

**Pull request merged**